### ISSUE TITLE
Can't start containers after upgrading lxd to 0.15

### ISSUE THREAD

**Issue by karlhaas** [lxc/lxd#1018]
Just upgraded my server (lxd 0.15 got installed). Now I can't start the containers anymore!
Is there a fix for this?

```
>whoami
root
>uname -a
Linux bob 3.13.0-61-generic #100-Ubuntu SMP Wed Jul 29 11:21:34 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
>lxc version
0.15
>lxc image list
+--------+--------------+--------+-------------+--------+------------------------------+
| ALIAS  | FINGERPRINT  | PUBLIC | DESCRIPTION |  ARCH  |         UPLOAD DATE          |
+--------+--------------+--------+-------------+--------+------------------------------+
| ubuntu | 1610f2023796 | no     |             | x86_64 | Aug 6, 2015 at 4:21pm (CEST) |
+--------+--------------+--------+-------------+--------+------------------------------+
>lxc info test
Name: test
Status: STOPPED
>lxc start test
error saving config file for the container failed
>cd /var/log/lxd/
>tail lxd.log 
t=2015-08-16T12:18:34+0200 lvl=info msg=handling method=GET url=/1.0/operations/b3d0f48b-5156-4557-a9c4-711aa90dca53/wait ip=@
t=2015-08-16T12:18:38+0200 lvl=info msg=handling method=GET url=/1.0 ip=@
t=2015-08-16T12:18:38+0200 lvl=info msg=handling method=PUT url=/1.0/containers/nginx/state ip=@
t=2015-08-16T12:18:38+0200 lvl=info msg=handling method=GET url=/1.0/operations/993d7a7c-e10b-4721-8d07-2738515615b6/wait ip=@
t=2015-08-16T12:35:27+0200 lvl=info msg=handling method=GET url=/1.0 ip=@
t=2015-08-16T12:35:27+0200 lvl=info msg=handling method=GET url="/1.0/containers/test?log=false" ip=@
t=2015-08-16T12:35:27+0200 lvl=info msg=handling method=GET url="/1.0/containers/test/snapshots?recursion=1" ip=@
t=2015-08-16T12:35:36+0200 lvl=info msg=handling method=GET url=/1.0 ip=@
t=2015-08-16T12:35:36+0200 lvl=info msg=handling method=PUT url=/1.0/containers/test/state ip=@
t=2015-08-16T12:35:36+0200 lvl=info msg=handling method=GET url=/1.0/operations/67a26a6e-a260-4059-814a-adcbffe28465/wait ip=@
>cd test/
>tail lxc.log
            lxc 1439721327.118 INFO     lxc_confile - confile.c:config_idmap:1433 - read uid map: type u nsid 0 hostid 100000 range 65536
            lxc 1439721327.118 INFO     lxc_confile - confile.c:config_idmap:1433 - read uid map: type g nsid 0 hostid 100000 range 65536
            lxc 1439721327.121 WARN     lxc_cgmanager - cgmanager.c:cgm_get:993 - do_cgm_get exited with error
            lxc 1439721327.123 WARN     lxc_cgmanager - cgmanager.c:cgm_get:993 - do_cgm_get exited with error
            lxc 1439721327.132 INFO     lxc_confile - confile.c:config_idmap:1433 - read uid map: type u nsid 0 hostid 100000 range 65536
            lxc 1439721327.132 INFO     lxc_confile - confile.c:config_idmap:1433 - read uid map: type g nsid 0 hostid 100000 range 65536
            lxc 1439721336.766 INFO     lxc_confile - confile.c:config_idmap:1433 - read uid map: type u nsid 0 hostid 100000 range 65536
            lxc 1439721336.766 INFO     lxc_confile - confile.c:config_idmap:1433 - read uid map: type g nsid 0 hostid 100000 range 65536
            lxc 1439721336.769 ERROR    lxc_utils - utils.c:get_template_path:1320 - No such file or directory - bad template: /tmp/lxd_lxc_startconfig_432461405
            lxc 1439721336.769 ERROR    lxc_container - lxccontainer.c:do_lxcapi_create:1339 - bad template: /tmp/lxd_lxc_startconfig_432461405
```

**Comment by karlhaas** [comment]
Also tried to download new versions of the images ... same error
```
>lxc image list
+---------+--------------+--------+-------------------------------------------------+--------+--------------------------------+
|  ALIAS  | FINGERPRINT  | PUBLIC |                   DESCRIPTION                   |  ARCH  |          UPLOAD DATE           |
+---------+--------------+--------+-------------------------------------------------+--------+--------------------------------+
| ubuntu  | 1610f2023796 | no     |                                                 | x86_64 | Aug 6, 2015 at 4:21pm (CEST)   |
| debian  | 68dc60dfcc72 | no     | debian wheezy x86_64 (default) (20150814_22:42) | x86_64 | Aug 16, 2015 at 12:50pm (CEST) |
| ubuntu2 | fef82be47702 | no     | ubuntu trusty x86_64 (default) (20150815_03:49) | x86_64 | Aug 16, 2015 at 12:46pm (CEST) |
+---------+--------------+--------+-------------------------------------------------+--------+--------------------------------+
>lxc launch debian test
Creating test done.
Starting test error.
error saving config file for the container failed
>tail /var/log/lxd/test/lxc.log 
            lxc 1439722138.815 WARN     lxc_cgmanager - cgmanager.c:cgm_get:993 - do_cgm_get exited with error
            lxc 1439722138.817 WARN     lxc_cgmanager - cgmanager.c:cgm_get:993 - do_cgm_get exited with error
            lxc 1439722138.827 INFO     lxc_confile - confile.c:config_idmap:1433 - read uid map: type u nsid 0 hostid 100000 range 65536
            lxc 1439722138.827 INFO     lxc_confile - confile.c:config_idmap:1433 - read uid map: type g nsid 0 hostid 100000 range 65536
            lxc 1439722267.448 INFO     lxc_confile - confile.c:config_idmap:1433 - read uid map: type u nsid 0 hostid 100000 range 65536
            lxc 1439722267.448 INFO     lxc_confile - confile.c:config_idmap:1433 - read uid map: type g nsid 0 hostid 100000 range 65536
            lxc 1439722271.726 INFO     lxc_confile - confile.c:config_idmap:1433 - read uid map: type u nsid 0 hostid 100000 range 65536
            lxc 1439722271.726 INFO     lxc_confile - confile.c:config_idmap:1433 - read uid map: type g nsid 0 hostid 100000 range 65536
            lxc 1439722271.727 ERROR    lxc_utils - utils.c:get_template_path:1320 - No such file or directory - bad template: /tmp/lxd_lxc_startconfig_648243392
            lxc 1439722271.727 ERROR    lxc_container - lxccontainer.c:do_lxcapi_create:1339 - bad template: /tmp/lxd_lxc_startconfig_648243392
```

**Comment by pcdummy** [comment]
Had that problem once to, fixed it by making sure that i have the LXC version from the PPA not any else.

**Comment by karlhaas** [comment]
I installed the version from the PPA. I tried to re-install but still the same problem ...
```
>apt-cache policy lxd
lxd:
  Installed: 0.15-0ubuntu1~ubuntu14.04.1~ppa1
  Candidate: 0.15-0ubuntu1~ubuntu14.04.1~ppa1
  Version table:
 *** 0.15-0ubuntu1~ubuntu14.04.1~ppa1 0
        500 http://ppa.launchpad.net/ubuntu-lxc/lxd-git-master/ubuntu/ trusty/main amd64 Packages
        100 /var/lib/dpkg/status
>lxc start test
error saving config file for the container failed
```

**Comment by POIMann** [comment]
Suffering the same issue here.
Please advise, thanks!

**Comment by pcdummy** [comment]
Not LXD but LXC, make sure you have the version from PPA.

**Comment by karlhaas** [comment]
yes, is from PPA
```
>apt-cache policy lxc
lxc:
  Installed: 1.1.3-0ubuntu1~ubuntu14.04.1~ppa1
  Candidate: 1.1.3-0ubuntu1~ubuntu14.04.1~ppa1
  Version table:
 *** 1.1.3-0ubuntu1~ubuntu14.04.1~ppa1 0
        500 http://ppa.launchpad.net/ubuntu-lxc/lxd-git-master/ubuntu/ trusty/main amd64 Packages
        100 /var/lib/dpkg/status
```

**Comment by stgraber** [comment]
Hello,

This breakage occurred due to the release of LXC 1.1.3 which fixes an ABI regression in the LXC 1.1 branch. As a result, LXD must be rebuilt for it to work again.

I've pushed a no-change rebuild of LXD to Ubuntu now which will then be automatically backported in all the relevant PPAs over the next 6 hours.

Sorry for the disruption.

**Comment by stgraber** [comment]
Closing this as the fix has now reached Ubuntu, auto-backports will happen when cron gets to it.
